nuxeo-distribution/nuxeo-server-hotreload-tests/src/test/java/org/nuxeo/ftest/server/hotreload/ITDevHotReloadTest.java

```diff
@@ -249,4 +260,51 @@ public void testHotReloadFileImporters() throws IOException {
         assertEquals("Foo", docType);
     }
 
+    /**
+     * @since 11.1
+     */
+    @Test
+    public void shouldComputeMimeTypeWhenBlobsSchemaIsModified() throws IOException {
+        // 1st deploy that define a new document type `CustomDocType` and its model.
+        hotReloadRule.deployJarDevBundle(ITDevHotReloadTest.class, "testHotReloadSchema");
+
+        // Creates the document and its content.
+        String docId = RestHelper.createDocument("/", "CustomDocType", "myDocument");
+        addBlobToDocument(docId, CUSTOM_IMAGE_PATH, "anyImage", "png");
+        addBlobToDocument(docId, "custom:question/image", "anyImage", "png");
+        addBlobToDocument(docId, "custom:question/audio", "anySong", "mp3");
+
+        // Check that all mime types are correctly computed.
+        Map<String, Object> properties = RestHelper.fetchDocumentProperties(docId, List.of("custom"));
+        Map<String, Object> image = (Map) properties.get(CUSTOM_IMAGE_PATH);
+        assertEquals("image/png", image.get(MIME_TYPE));
+
+        Map<String, Map<String, Object>> question = (Map) properties.get(CUSTOM_QUESTION_PATH);
+        assertEquals("image/png", question.get("image").get(MIME_TYPE));
+        assertEquals("audio/mpeg", question.get("audio").get(MIME_TYPE));
+
+        // 2nd deploy that override the exiting document model by removing the `question:image` property.
+        hotReloadRule.deployJarDevBundle(ITDevHotReloadTest.class, "testHotReloadUpdateSchema");
+
+        // Ensure that the event handler responsible of computing the mime type is correctly called when
+        // the schema is updated.
+        addBlobToDocument(docId, CUSTOM_IMAGE_PATH, "newImage", "psd");
+
+        // Check then the new mime type is correctly computed.
+        properties = RestHelper.fetchDocumentProperties(docId, List.of("custom"));
+        image = (Map) properties.get(CUSTOM_IMAGE_PATH);
+        question = (Map) properties.get(CUSTOM_QUESTION_PATH);
+        assertEquals("application/photoshop", image.get(MIME_TYPE));
+        assertEquals("audio/mpeg", question.get("audio").get(MIME_TYPE));
+        assertNull(question.get("image"));
+
+        RestHelper.addDocumentToDelete(docId);
+    }
+
+    /** @since 11.1 **/
+    protected void addBlobToDocument(String docId, String xpath, String fileName, String extension) throws IOException {
+        File file = Files.createTempFile(fileName, ".".concat(extension)).toFile();
```

Comment:
Here just use simple concatenation:
```suggestion
        File file = Files.createTempFile(fileName, "." + extension).toFile();
```